staticcheck/doc.go

```diff
@@ -332,6 +332,14 @@ the various parsing and formatting functions in \'strconv\'.`,
 		MergeIf:  lint.MergeIfAny,
 	},
 
+	"SA1031": {
+		Title:    `Overlapping byte slices passed to an encoder`,
+		Text:     `In an encoding function of the form \'Encode(dst, src)\', \'dst\' and \'src\' were found to reference the same memory. This can result in \'src\' bytes being overwritten before they are read, when the encoder writes more than one byte per \'src\' byte.`,
+		Since:    "Unreleased",
+		Severity: lint.SeverityError,
```

Comment:
I think this should be a warning, not an error, because the code doesn't panic or return an error.